### ISSUE TITLE
fuel-explorer does not compile to native executor

### ISSUE THREAD

**Issue by lostman** [FuelLabs/fuel-indexer#1091]
I tried compiling `fuel-explorer` to a native executor to debug #1090.

However, it can't be done due to the use of `get_or_create` in `impl From`.

https://github.com/FuelLabs/fuel-indexer/blob/68d0690cd9e2d2dd32fd8ec58d99ea7d30f37a81/examples/fuel-explorer/fuel-explorer/src/lib.rs#L134

For native execution, it is an `async` function.

**Comment by ra0x3** [comment]
@lostman 
- So you would have to make several changes to the the explorer in order to make it compile natively
- The `get_or_create` you mention isn't async because the current explorer was created for WASM execution
- If you hypothetically wanted to create an explorer with native execution you would:
  - `forc index new explorer --namespace fuel --native`
  - Copy over the schema, manifest, and lib module from the WASM explorer to this new explorer
  - Update all `.save()`, `.get_or_create()` with `.await` and all functions `fn` with `async`

TLDR: Compiling the explorer with native execution would require several changes to the WASM explorer :)

**Comment by ra0x3** [comment]
@lostman Per my most recent comment gonna close this, but if you feel like I'm closing this in error, feel free to open it back up :)
